### ISSUE TITLE
[Feature] - Svelte 5 support

### ISSUE THREAD

**Issue by josdejong** [RobBrazier/svelte-awesome#1186]
I'm upgrading a Svelte library of mine that uses `svelte-awesome` to support Svelte 5. As far as I can see, `svelte-awesome` works just fine with Svelte 5. However, in the `package.json`, the libary hasn't yet listed Svelte 5 as a peer dependency:

https://github.com/RobBrazier/svelte-awesome/blob/6b98100ec4a9ac8f02bc4ec0aaa8ff4097779454/package.json#L54-L56

Due to this, `npm install` fails with the following error:

```bash
$ npm install

npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: svelte-awesome@3.3.4
npm error Found: svelte@5.0.0-next.264
npm error node_modules/svelte
npm error   dev svelte@"5.0.0-next.264" from the root project
npm error   peer svelte@"^4.0.0 || ^5.0.0-next.0" from @sveltejs/kit@2.7.0
npm error   node_modules/@sveltejs/kit
npm error     dev @sveltejs/kit@"2.7.0" from the root project
npm error     peer @sveltejs/kit@"^2.0.0" from @sveltejs/adapter-auto@3.2.5
npm error     node_modules/@sveltejs/adapter-auto
npm error       dev @sveltejs/adapter-auto@"3.2.5" from the root project
npm error   10 more (@sveltejs/package, @sveltejs/vite-plugin-svelte, ...)
npm error
npm error Could not resolve dependency:
npm error peer svelte@"^3.43.1 || ^4.0.0" from svelte-awesome@3.3.4
npm error node_modules/svelte-awesome
npm error   svelte-awesome@"^3.3.4" from the root project
npm error
npm error Conflicting peer dependency: svelte@4.2.19
npm error node_modules/svelte
npm error   peer svelte@"^3.43.1 || ^4.0.0" from svelte-awesome@3.3.4
npm error   node_modules/svelte-awesome
npm error     svelte-awesome@"^3.3.4" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```

As a workaround, we can use `npm install --force`, but I would love to see `svelte-awesome` to officially support Svelte 5.

**Comment by RobBrazier** [comment]
sorry I must have missed this notification! I tried this before and it failed miserably still on the latest v5 at the time, not sure whether it's because the version is still pre-release though

could you try with version 3.3.3? (yes an older version)

**Comment by josdejong** [comment]
Yeah it's tricky. I see in `svelte-awesome@3.3.3` you had defined peerDependencies as `"svelte": ">= 3.43.1 < 6"`. That will work as soon as svelte v5 is released, but will not match release candidates. I tested it that with the official [semver](https://www.npmjs.com/package/semver) package: 

```js
console.log(semver.satisfies("5.0.0-next.264", ">= 3.43.1 < 6")) // false
console.log(semver.satisfies("5.0.0", ">= 3.43.1 < 6")) // true
```

In a package like [`svelte2tsx`](https://github.com/sveltejs/language-tools/blob/master/packages/svelte2tsx/package.json#L45-L48) I see an approach of defining all major versions and release candidates one by one, and that seems to work (and it is explicit and well readable to me too):

```js
console.log(semver.satisfies("5.0.0-next.264", ">= 3.43.1 || >= 4.0.0 || >= 5.0.0-next || >= 5.0.0")) // true
```
Is that an idea?

**Comment by mudiageo** [comment]
@RobBrazier Now that Svelte 5 has been released, perhaps you can now update peerDependencies to include Svelte 5

**Comment by RobBrazier** [comment]
sorry not had much time recently, this is now released in 3.3.5

**Comment by josdejong** [comment]
Thanks Rob for the latest release! And for building and maintaining this _awesome_ libary 😉
